feat: add getting started page for knative functions (#201)

lstocchi opened the pull request:
it resolves #201 
this requires https://github.com/redhat-developer/intellij-common/pull/156 to work

@mohitsuman give it a look at the experience and the content itself. 

The UI is similar to the learning feature plugin made by IJ. The only difference is the structure of a lesson. I made it similar to what's in VSCode.
![getting-started-demo](https://user-images.githubusercontent.com/49404737/182729825-df9fcf3b-0534-4dc0-8027-2cefa95897f9.gif)

cc @sudhirverma

jeffmaury reviewed (approved):
LGTM but would be good to have a kind of course version and at startup display automatically the course if user never saw it or if course version changed. WDYT ?

lstocchi commented:
> LGTM but would be good to have a kind of course version and at startup display automatically the course if user never saw it or if course version changed. WDYT ?

Added the course version... the issue is how to show the toolwindow when necessary.
The `isDoNotShowAtActivation` method is deprecated and replaced with a property to be set in the extensionPoint in the plugin.xml. I guess we cannot use it as the calculation is done at runtime, right?

Found a hack with the toolwindowmanagerlistener ... not that convince me much but this is the only thing found so far -> https://github.com/redhat-developer/intellij-knative/pull/210/commits/4a8f0b17cbc6fc598fa525be1682a5afac994398#diff-3b241b322e3e3fd9f7b3cc3f53eef2b9da6988cec0fc442084ce1851d5d6d3e5R41

jeffmaury reviewed (approved):
LGTM

mohitsuman commented:
@lstocchi For the Getting Started, can we make the gifs bigger and maybe increase resolution? I think that way we can clearly focus on the action and not the whole IJ screen.

mohitsuman commented on src/main/java/com/redhat/devtools/intellij/knative/ui/toolwindow/GettingStartedToolWindow.java:
```diff
@@ -0,0 +1,148 @@
+                "<html><p>Building a function allows to create an image from the source code and being able to run and/or deploy it.</p>" +
+                        "<p>Only a function which has the source code opened in the IDE can be built. Right click on the function you want to build" +
+                        "(look for the right node in the Functions tree), " +
+                        "open the context menu (right-click on the node) and click on \"Build\". " +
```
```suggestion
                        "Open the context menu (right-click on the node) and click on \"Build\". " +
```

mohitsuman commented on src/main/java/com/redhat/devtools/intellij/knative/ui/toolwindow/GettingStartedToolWindow.java:
```diff
@@ -0,0 +1,148 @@
+        URL gifRunFuncLesson = getLessonGif("run-func-lesson.gif");
+        GettingStartedLesson runFuncLesson = new GettingStartedLesson(
+                "Run function",
+                "<html><p>Run your function to see how it works locally before to push it on your cluster.</p>" +
```
```suggestion
                "<html><p>Run the function locally on the current directory opened in the IDE.</p>" +
```

mohitsuman commented on src/main/java/com/redhat/devtools/intellij/knative/ui/toolwindow/GettingStartedToolWindow.java:
```diff
@@ -0,0 +1,148 @@
+        GettingStartedLesson runFuncLesson = new GettingStartedLesson(
+                "Run function",
+                "<html><p>Run your function to see how it works locally before to push it on your cluster.</p>" +
+                        "<p>Only a function which has the source code opened in the IDE can be run. " +
```
```suggestion
                 
```

mohitsuman commented on src/main/java/com/redhat/devtools/intellij/knative/ui/toolwindow/GettingStartedToolWindow.java:
```diff
@@ -0,0 +1,148 @@
+                "Run function",
+                "<html><p>Run your function to see how it works locally before to push it on your cluster.</p>" +
+                        "<p>Only a function which has the source code opened in the IDE can be run. " +
+                        "Right click on the function you want to run (look for its node in the Functions tree), open the context menu (right-click on the node) " +
```
```suggestion
                        "Open the context menu (right-click on the node) " +
```

mohitsuman commented on src/main/java/com/redhat/devtools/intellij/knative/ui/toolwindow/GettingStartedToolWindow.java:
```diff
@@ -0,0 +1,148 @@
+        URL gifDeployFuncLesson = getLessonGif("deploy-func-lesson.gif");
+        GettingStartedLesson deployFuncLesson = new GettingStartedLesson(
+                "Deploy function",
+                "<html><p>Deploy your function to your cluster directly from your preferred IDE.</p>" +
```
```suggestion
                "<html><p>Deploys a function to the currently configured Knative-enabled cluster.</p>" +
```
